enhancements/update/cvo-log-level-api.md

```diff
@@ -0,0 +1,209 @@
+
+```go
+// +kubebuilder:validation:Enum="";Normal;Debug;Trace;TraceAll
+type CVOLogLevel string
+
+var (
+	// Normal is the default.  Normal, working log information, everything is fine, but helpful notices for auditing or common operations.  In kube, this is probably glog=2.
+	CVOLogLevelNormal CVOLogLevel = "Normal"
+
+	// Debug is used when something went wrong.  Even common operations may be logged, and less helpful but more quantity of notices.  In kube, this is probably glog=4.
+	CVOLogLevelDebug CVOLogLevel = "Debug"
+
+	// Trace is used when something went really badly and even more verbose logs are needed.  Logging every function call as part of a common operation, to tracing execution of a query.  In kube, this is probably glog=6.
+	CVOLogLevelTrace CVOLogLevel = "Trace"
+
+	// TraceAll is used when something is broken at the level of API content/decoding.  It will dump complete body content.  If you turn this on in a production cluster
+	// prepare from serious performance issues and massive amounts of logs.  In kube, this is probably glog=8.
+	CVOLogLevelTraceAll CVOLogLevel = "TraceAll"
+)
+```
+
+### Workflow Description
+
+Given a cluster administrator and a working cluster for which the administrator is responsible.
+
+**cluster administrator** is a human user responsible for managing the cluster.
+
+1. The cluster administrator notices an issue in the cluster and chooses to troubleshoot the issue.
+2. The cluster administrator after some troubleshooting notices that the logs of the Cluster Version Operator (CVO) might help.
+3. The cluster administrator notices that the logs are not detailed enough to troubleshoot the issue, and the administrator raises the log level from the default value `Normal` to the level `Trace` by simply modifying the ClusterVersion object via the web console or by patching the resource by using the CLI.
+4. The cluster administrator fixes the issue.
+5. The cluster administrator notices that the CVO outputs too many logs for the administrator's liking.
+6. The cluster administrator lowers the log level of the CVO to the level `Normal`.
+7. The cluster administrator is now a happy cluster administrator.
+
+#### Variation [optional]
+
+N/A
+
+### API Extensions
+
+See the section [Proposal](#proposal) for the proposed API changes.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The CVO needs to change the log level dynamically without redeploying itself. The CVO currently uses the package [`klog`](https://github.com/kubernetes/klog) for logging. This package does not provide a simple way of changing the log level dynamically. However, it is possible, and there already exists a function 
+[`SetLogLevel`](https://github.com/openshift/library-go/blame/8477ec72b72554c01afff4df9c6a90c0e492ea87/pkg/operator/loglevel/util.go#L63-L103) in the [`openshift/library-go`](https://github.com/openshift/library-go) repository to remedy this issue.
+
+#### Hypershift [optional]
```

Comment:
Hello @enxebre, @csrwng, @sjenning :wave: 

This enhancement proposes to add a log level API change to the ClusterVersion resource.

After the enhancement is merged, the API change is merged, and the implementation is done in the [CVO](https://github.com/openshift/cluster-version-operator) repository, a hosted cluster administrator could set an undesirable log level for its hosted CVO by modifying this new field in the hosted ClusterVersion. This enhancement proposes to fix this by implementing a reconciliation of a fixed value in the below-mentioned section in the HyperShift codebase after the API change is merged and before the implementation in the CVO repository is done.

Is this proposition for the HyperShift use case acceptable?